pennylane/fourier/reconstruct.py

```diff
@@ -616,15 +617,14 @@ def circuit(x, Y, f=1.0):
 
     atol = 1e-8
     ids, recon_fn, jobs, need_f0 = _prepare_jobs(ids, nums_frequency, spectra, shifts, atol)
-    arg_names = list(signature(qnode.func).parameters.keys())
+    sign_fn = qnode.func if isinstance(qnode, qml.QNode) else qnode
```

Comment:
this is a small modification to allow passing qfuncs directly?

Reply:
Yes indeed.

In particular, it allows passing python functions :D Previously, only QNodes generated with the decorator where covered, I believe. (Maybe QNodes generated with the class initialization also but certainly not python callables, because they don't have the `func` attribute.